client/app/components/ParameterMappingInput.jsx

```diff
@@ -0,0 +1,267 @@
+          {
+            (existingParamNames.length > 0) &&
+            <Option value={MappingType.DashboardMapToExisting}>Map to existing parameter</Option>
+          }
+          <Option value={MappingType.StaticValue}>Use static value for the parameter</Option>
+          <Option value={MappingType.WidgetLevel}>Keep the parameter at the widget level</Option>
+        </Select>
+      </div>
+    );
+  }
+
+  renderDashboardAddNew() {
+    const { mapping, existingParamNames } = this.props;
+    const alreadyExists = includes(existingParamNames, mapping.mapTo);
+    return (
+      <div className={'m-t-10' + (alreadyExists ? ' has-error' : '')}>
+        <input
+          type="text"
+          className="form-control"
+          value={mapping.mapTo}
+          onChange={event => this.updateParamMapping(mapping, { mapTo: event.target.value })}
+        />
+        { alreadyExists &&
+        <div className="help-block">
+          Dashboard parameter with this name already exists
+        </div>
+        }
+      </div>
+    );
+  }
+
+  renderDashboardMapToExisting() {
+    const { mapping, existingParamNames } = this.props;
+    return (
+      <div className="m-t-10">
+        <Select
+          className="w-100"
+          defaultValue={mapping.mapTo}
+          onChange={mapTo => this.updateParamMapping(mapping, { mapTo })}
+          disabled={existingParamNames.length === 0}
+          dropdownClassName="ant-dropdown-in-bootstrap-modal"
+        >
+          {map(existingParamNames, name => (
+            <Option value={name} key={name}>{ name }</Option>
+          ))}
+        </Select>
+      </div>
+    );
+  }
+
+  renderStaticValue() {
+    const { mapping } = this.props;
+    return (
+      <div className="m-t-10">
+        <ParameterValueInput
+          className="w-100"
+          type={mapping.param.type}
+          value={mapping.value}
+          enumOptions={mapping.param.enumOptions}
+          queryId={mapping.param.queryId}
+          onSelect={value => this.updateParamMapping(mapping, { value })}
+          clientConfig={this.props.clientConfig}
+          Query={this.props.Query}
+        />
+      </div>
+    );
+  }
+
+  renderInputBlock() {
+    const { mapping } = this.props;
+    switch (mapping.type) {
+      case MappingType.DashboardAddNew: return this.renderDashboardAddNew();
+      case MappingType.DashboardMapToExisting: return this.renderDashboardMapToExisting();
+      case MappingType.StaticValue: return this.renderStaticValue();
+      // no default
+    }
+  }
+
+  renderTitleInput() {
+    const { mapping } = this.props;
+    return (
+      <div className="m-t-10">
+        <input
+          type="text"
+          className="form-control"
+          value={mapping.title}
+          onChange={event => this.updateParamMapping(mapping, { title: event.target.value })}
+          placeholder="Change parameter title (leave empty to use existing"
```

Comment:
The placeholder text is too long and gets cut off. My suggestion is:

1. Let's add labels for all the inputs in this dialog. It will solve this issue but also solve the issue of not knowing what the input is for after entering the value.
2. We should show the current parameter title -- either as placeholder here or next to the parameter keyword.